docs/swagger2.yml

```diff
@@ -27,673 +19,414 @@ consumes:
           type: string
-      $ref: '#/definitions/MemoryResource'
-      $ref: '#/definitions/CPUResource'
-      $ref: '#/definitions/TimeoutResource'
-      $ref: '#/definitions/IdleTimeoutResource'
-      # TODO(?) annotations? 
+      annotations:
+        type: object
+        description: Application annotations - this is a map of annotations attached to this app, keys must not exceed 128 bytes and must consist of non-whitespace printable ascii characters, and the seralized representation of individual values must not exeed 512 bytes
+        additionalProperties:
+          type: object
       created_at:
         type: string
         format: date-time
-        description: Time when function was created. Always in UTC RFC3339.
+        description: Time when app was created. Always in UTC.
         readOnly: true
       updated_at:
         type: string
         format: date-time
-        description: Most recent time that function was updated. Always in UTC RFC3339.
+        description: Most recent time that app was updated. Always in UTC.
         readOnly: true
 
-  Trigger:
+  Fn:
     type: object
     properties:
-      app_id:
-        type: string
-        description: App ID
       id:
         type: string
-        description: App ID
+        description: UUID
         readOnly: true
       name:
         type: string
-        description: unique name for this trigger, used to identify this trigger
-      type:
+        description: unique name for this function.
+      app_name:
+        $ref: '#/parameters/AppName'
+      image:
         type: string
-        description: class of trigger, e.g. schedule, http
-      source:
+        description: "full container image name, e.g. hub.docker.com/fnproject/yo or fnproject/yo (default registry: hub.docker.com)"
+      image_id:
         type: string
-        description: uri path for this trigger. e.g. `sayHello`, `say/hello`
-      func:
+        description: "docker image id, sha hash"
+      revision:
         type: string
-        description: "fully qualified function name referencing a function in a repository. eg: `funchub.fnproject.io/jimbo/somefunc:1.2.3`"
-      $ref: '#/definitions/MemoryResource'
-      $ref: '#/definitions/CPUResource'
-      $ref: '#/definitions/TimeoutResource'
-      $ref: '#/definitions/IdleTimeoutResource'
+        description: "server generated version denoting image_id and configuration version."
       config:
         type: object
-        description: Trigger configuration - overrides application configuration
-        additionalProperties:
-          type: string
-      annotations:
-        type: object
-        description: Trigger annotations - this is a map of annotations attached to this trigger, keys must not exceed 128 bytes and must consist of non-whitespace printable ascii characters, and the seralized representation of individual values must not exeed 512 bytes
-        additionalProperties:
-          type: object
```

Comment:
config and annotations should remain here I think, and the resource config stuff below should be top level on the func (rather than in the `config:` section